src/main/java/org/sonarlint/intellij/SonarLintIntelliJClient.kt

```diff
@@ -374,8 +376,14 @@ object SonarLintIntelliJClient : SonarLintClient {
     override fun selectProxies(params: SelectProxiesParams): CompletableFuture<SelectProxiesResponse> {
         val uri = URI.create(params.uri)
         val proxiesResponse =
-            SelectProxiesResponse(CommonProxy.getInstance().select(uri).stream().map { ProxyDto(it.type(), uri.host, uri.port) }
-                .collect(Collectors.toList()))
+            SelectProxiesResponse(CommonProxy.getInstance().select(uri).stream().map {
+                if (it.type() != Proxy.Type.DIRECT) {
+                    val socketAddress = it.address() as InetSocketAddress
+                    ProxyDto(it.type(), socketAddress.hostString, socketAddress.port)
+                } else {
+                    ProxyDto.NO_PROXY
```

Comment:
On which occasion can a proxy not be a proxy? I have to say I find this quite irritating that it actually happened. The rest of the changes make sense 👍

Reply:
Ah it's already like this on JDK side, so nvm.